avro-fastserde/src/main/java/com/linkedin/avro/fastserde/PrimitiveFloatList.java

```diff
@@ -0,0 +1,291 @@
+package com.linkedin.avro.fastserde;
+
+import java.io.IOException;
+import java.util.AbstractList;
+import java.util.Collection;
+import java.util.Iterator;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericArray;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.io.Decoder;
+
+
+/**
+ * This is a re-implementation of Avro's {@link GenericData.Array} class.
+ *
+ * Compared to the Avro implementation, it offers the following GC-related optimizations:
+ *
+ * - It does not, by default, box primitive floats into Object Floats, though it will still do so if the
+ *   regular functions are called (e.g.: {@link #get(int)}, for compatibility purposes. In order to avoid
+ *   boxing, the {@link #getPrimitive(int)} function can be used instead.
+ *
+ * - It does not maintain a reference to a {@link Schema} instance, since that schema would always be the
+ *   same. Instead, it defines a static {@link #SCHEMA} which is used by all instances.
+ *
+ * - It re-implements {@link #compareTo(GenericArray)}, {@link #equals(Object)} and {@link #hashCode()}
+ *   in order to leverage the primitive types, rather than causing unintended boxing.
+ *
+ */
+public class PrimitiveFloatList extends AbstractList<Float>
+    implements GenericArray<Float>, Comparable<GenericArray<Float>> {
+  private static final float[] EMPTY = new float[0];
+  private static final Schema FLOAT_SCHEMA = Schema.create(Schema.Type.FLOAT);
+  private static final Schema SCHEMA = Schema.createArray(FLOAT_SCHEMA);
+
+  private int size;
+  private float[] elements = EMPTY;
+
+  public PrimitiveFloatList(int capacity) {
+    if (capacity != 0) {
+      elements = new float[capacity];
+    }
+  }
+
+  public PrimitiveFloatList(Collection<Float> c) {
+    if (c != null) {
+      elements = new float[c.size()];
+      addAll(c);
+    }
+  }
+
+  /**
+   * Instantiate (or re-use) and populate a {@link PrimitiveFloatList} from a {@link org.apache.avro.io.ResolvingDecoder}.
```

Comment:
Any `Decoder` will do, and in fact, we stand to gain from NOT using a `ResolvingDecoder`. We should update the JavaDoc link accordingly.

Reply:
Done.